### ISSUE TITLE
[EYE Data] ‘No data’ categories count towards abnormal phenotypes in the stats pipeline leading to incorrect calls

### ISSUE THREAD

**Issue by ficolo** [mpi2/impc_stats_pipeline#61]
‘No data’ categories count towards abnormal phenotypes in the stats pipeline leading to incorrect calls:

https://www.mousephenotype.org/data/charts?accession=MGI:3646469&allele_accession_id=MGI:5882076&zygosity=homozygote&parameter_stable_id=JAXLA_EYE_018_001&pipeline_stable_id=JAXLA_001&procedure_stable_id=JAXLA_EYE_001&parameter_stable_id=JAXLA_EYE_018_001&phenotyping_center=JAX.

It was agreed with the morphology WG that all ‘no data’ categories should be excluded from the analysis. DR7 notes indicate that this was fixed.

There has been some discussion in the past about how to collapse the categories with left and right eye being considered separately (for stats and the graph/table). Currently the stats seem to be working on normal v. all the others as with other categorical data (activity (body position): https://www.mousephenotype.org/impress/OntologyInfo?action=list&procID=797#38129 , data: https://www.mousephenotype.org/data/charts?accession=MGI:3583900&allele_accession_id=MGI:5548895&zygosity=homozygote&parameter_stable_id=IMPC_CSD_029_001&pipeline_stable_id=HRWL_001&procedure_stable_id=IMPC_CSD_003&parameter_stable_id=IMPC_CSD_029_001&phenotyping_center=MRC%20Harwell)

Some places to start looking into the stats pipelines behaviour for this issue:
 https://github.com/search?q=repo%3Ampi2%2Fimpc_stats_pipeline%20no%20data%20for%20both%20eyes&type=code

**Comment by marinak-ebi** [comment]
It looks like this problem is more complicated than originally thought.

In pipeline, there are two files which do something with categories regarding their names or p-value calculation:

1. `CategoryMap.conf` renames category values. It does this globally for all parameters.
2. `CategoryRemapping.tsv`, marks for each category value whether it should be considered for p-value calculation. It is parameter specific, so because of this the file is quite long.

I have compared both files with the actual category values present in SOLR. It looks like of those files are seriously obsolete, that is, they don't contain the up to date category names. I have verified that this problem affects both early and late adult data.

To fix this properly, I'll need to fetch all category values for all eye phenotypes, curate them, and update `CategoryRemapping.tsv`. There are 60+ parameters and 20+ possible category values between them, so it'll take some time.

Renaming presents another problem, because it is being done globally. While we can rename eye specific phenotypes ("present both eyes" → "abnormal"), I'm not sure if we can globally rename "absent" to "normal", because for some non-eye phenotypes "absent" could mean something else, and as I mentioned the renaming is global.
